lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1734,9 +1734,24 @@ class MongoClientInterface {
     internalListObject(bucketName, params, extension, vFormat, log, cb) {
         const c = this.getCollection(bucketName);
         const getLatestVersion = this.getLatestVersion;
-        const cbOnce = jsutil.once(cb);
         let stream;
         let baseStream;
+        let resolvePhdKey;
+        const cbOnce = jsutil.once((err, data) => {
+            if (resolvePhdKey) {
+                resolvePhdKey.removeAllListeners();
+                resolvePhdKey.destroy();
+            }
+            if (baseStream) {
+                baseStream.removeAllListeners();
+                baseStream.destroy();
+            }
+            if (stream) {
```

Comment:
in some case (Bucket v0 case), `baseStream` may be the same as `stream` : so we should handle this in the callback to avoid using the destroyed stream...

Reply:
I don't see any issue, as in nodejs, destroying an already destroyed stream is a no-op...